# Final Project Report


## *Introduction*
Responding to changing glucose levels is a requirement found in all eukaryotic systems. Processing, storing, and utilizing incoming nutrients from the environment is required to maintain homeostatic levels of cellular function across all eukaryotes. It is critical to maintain a normal amount of sugar in the blood, by storing away any excess sugar or utilizing that storage or creating glucose upon instances where there is too little glucose within the blood (Chourpiliadis and Mohiuddin, 2019; Kolb et al., 2020). As all cell types depend on glucose, the regulation of glucose levels is critical. As such, this process is well conserved from mammals to invertebrates, such as *Drosophila melanogaster (Drosophila)* (Das and Dobens, 2015; Mattila and Hietakangas, 2017). 
*Drosophila* regulate and respond to changing sugar levels similarly to mammalian systems, possessing several homologs and orthologs of transcription factors that are involved in regulating various metabolic genes. However, unlike mammalian systems, it has been well established that the Transforming Growth Factor-Beta (TGF-β) superfamily plays an essential role in the secretion of insulin, an important protein required to signal for cells to uptake glucose (Ghosh and O’Connor, 2014). The TGF-β ligand, Dawdle, has been shown in studies to be required for insulin secretion, seen by an accumulation of unreleased insulin in its absence (Ghosh and O’Connor, 2014). *dawdle* mutants develop diabetes-like phenotypes due to the lack of insulin secretion, such as an increase in blood sugar levels, obesity, and poor storage of glucose and glycogen (Ghosh and O’Connor, 2014; Graham and Pick, 2017). 
Because of its important role in maintaining sugar homeostasis, I was interested in identifying factors that may regulate dawdle expression. *Drosophila* have a small-network of genes that have been well-characterized in the response to increasing sugar levels in the diet, which are well-conserved to mammalian systems. For example, Mondo and Mlx (ChREBP/MondoA and Mlx in mammals) are two known master transcriptional regulators that are directly activated by the presence of sugar in the cell, leading to the activation of other downstream factors that will regulate genes involved in metabolism (Mattila et al., 2015; Mattila and Hietakangas, 2017). However, Mondo and Mlx are the only established transcription factors that regulate other sugar-responsive factors, and there are potentially unidentified factors that act independently of Mondo and Mlx. To try and identify these factors, I analyzed pre-existing RNA-Sequencing data from Mlx mutants and wild-type larvae to identify factors that are up-regulated under both conditions on high-sugar diets. 
## *Methods*
*Note: all commands used will be uploaded in a separate document.* 
## *Acquiring Data*
For both datasets, I started with downloading the data to check the quality of the reads. This was done utilizing the Linux command line and importing the raw data into a specified directory. Original data files were imported either using the “wget” command, or the sra toolkit.
## *Quality Check and Trimming of Data*
FastQC was ran to check the raw data quality. Fastp was used to trim any adapters or poor-quality reads. Read quality was assessed using the summary files from the fastp command, as well as comparing FastQC files before and after the trimming of reads. All further analysis utilized the trimmed sequence files.  
## *Alignment and Counting of Reads*
To perform the alignment, the reference genome used was the dmel_r6.48 assembly, which was released in 2022. This assembly was chosen because it was used for the alignment in the latest dataset that was published. The FASTA file for the genome was downloaded and a md5 checksum was performed to ensure that the data was not altered or corrupted. After performing the checksum, the reference genome was then indexed. The trimmed FASTQ sequences were then aligned to the reference genome using the Burrows-Wheeler Aligner (BWA). 
To count the number of reads, featureCounts was used. Reads counted included those that aligned to an exon in the genome. Counts were saved in a text file. Additionally, gene identifiers and reads were extracted into a separate file to be used for statistical analysis. 
## *Statistical Analysis*
All statistical analysis was done utilizing R Studio. Files containing the gene identifiers and counts were imported and transformed into a data frame. These files were then used to create a count matrix to use for Differential Expression analysis using a Negative Binomial distribution (DESeq) analysis. A meta-table was created with included replicates, genotype, and diet. Using the package DESeq2, DESeq analysis was performed, and significant differentially expressed genes were identified by having a p-value that was less than 0.05. These genes were then saved as a new data frame to analyze up- and down-regulation. For the Mattila et al., 2015 dataset, genes which were up-regulated on a high-sugar diet in a mlx mutant background were saved to a separate data frame for comparison. Analysis for the van dne Berg et al., 2023 dataset was done similarly. The two datasets of significant genes (from the mlx mutant background) were then merged, creating a new dataset of genes that overlapped between the two. Known and candidate Drosophila transcription factors were imported from FlyTF into a new data frame. This transcription factor data frame was then used to subset the shared genes for differentially expressed genes that encoded for known and candidate transcription factors. 
## *Plotting*
All heatmaps were made using R. Primarily, pheatmap was used.

## *Results*
*Note: all tables and heatmaps referred to will be appended to GitHub.*
From the Mattila dataset, the p-adjusted value used to determine significance was 0.05. From the DESeq analysis, differentially expressed genes on a high-sugar diet in a mlx mutant background were identified (see Significant_Genes_HSD_Diet_mlx1.xlsx). From these genes, transcription factors were identified using an imported dataset from FlyTF, which was then used to subset the significant genes by transcription factor (see TFs_HSD_Data_mlx1.xlsx, [Significant Genes HSD Diet mlx1](https://github.com/guanyinstevens/ee282/blob/final_project/Significant_Genes_HSD_Diet_mlx1.xlsx)). The only significant sugar-responsive transcription factor identified was grain.   
Additionally, the top 100 significant genes were identified in the Mattila dataset by performing the analysis solely by diet. This was done by using the head function in R to solely look at the top 100 genes sorted by p-value. The only known sugar-responsive gene identified was grain (grn). 
Similar analysis was done with the van den Berg et al., dataset and the DESeq analysis was done based on diet conditions to look at significant differentially expressed genes (genes with a p-adjusted value of < 0.05). These significant genes were then compared to the genes identified in the Mattila dataset, and similar genes were then identified by using the merge function in R. Genes that encode transcription factors were then subsetted, generating a list of 34 shared transcription factors across both datasets (see shared_tf_table.csv, [Shared TF Table](https://github.com/guanyinstevens/ee282/blob/final_project/shared_tf_table.csv)). Up- and down-regulation of these transcription factors were visualized using a heatmap (see Heatmap_of_shared_TFs.png, ![Heatmap of shared TFs](https://github.com/guanyinstevens/ee282/blob/final_project/Heatmap_of_shared_TFs.png)). The log2Fold changes were not scaled to capture the absolute expression levels of the transcription factors, as well as the biological variance in expression. 
## *Discussion*
*Drosophila* have a small network of genes that have been well-characterized in the response to increasing sugar levels in the diet, which are well-conserved to mammalian systems. However, while TGF-Beta signaling has been well characterized in its importance to maintaining sugar homeostasis in Drosophila, there has been no clear evidence of its role in sugar homeostasis in mammalian systems. The ligand that is responsible in regulating insulin secretion is Dawdle and the goal of my project was to identify potential transcription factors that regulate dawdle expression. Additionally, I wanted to identify factors that may be sugar responsive but work independently of the transcription factors Mondo and Mlx, which are two known master transcriptional regulators that are directly activated by the presence of sugar in the cell. To identify what transcription factors could regulate dawdle expression in Drosophila, I compared RNA-Sequencing data, specifically looking at genes that were significantly down- and up-regulated when larvae were challenged on a high-sugar diet. I compared these genes between two datasets, one in a *mlx* mutant background and the other in a wild-type background, with the goal of identifying transcription factors that remained up-regulated in both populations. The *mlx* mutant background was sufficient to identify genes that work independently of both Mondo and Mlx, as the two transcription factors work as a complex to regulate gene expression. 
To identify significant genes that were differentially expressed, I focused on the *mlx* mutant background, and performed DESeq using the diet parameters to identify genes that were significantly over-expressed or under-expressed on a high sugar diet. I performed a similar analysis for the second dataset that had wild-type samples. I then merged those datasets together, effectively creating a new dataset of genes that were significant in both datasets. From this, I identified genes that encoded for transcription factors and looked at their absolute expression levels by creating a heatmap. From this analysis, I identified 34 transcription factors that were significant in both datasets, and identified candidate transcription factors that could regulate dawdle expression (see shared_tf_table.csv, [shared_tf_table.csv](https://github.com/guanyinstevens/ee282/blob/final_project_report/shared_tf_table.csv)). 
Of these 34 transcription factors, genes that were up-regulated in both datasets that are known to be sugar-responsive included *mondo* and *grain (grn)*. While Mondo and Mlx do work as a complex, the initiation of the complex depends on Mondo directly binding to phosphorylated glucose (glucose-6-phosphate) or other hexoses, which then allows it to bind to Mlx. It is unknown what initially transcribes Mondo and Mlx, they become up-regulated when sugar levels are high and in turn activate genes that are involved in fat storage and metabolism. Grain is a direct downstream target of the Mondo-Mlx target and contributes to the expression of lipogenic genes on a low sugar diet, activation of glycolytic genes (Kokki et al. 2021). However, as it is up-regulated in the mlx mutant background, it could be suggested that grain could also work independently of Mondo-Mlx. Interestingly, a known sugar-responsive transcription factor, *sugarbabe (sug)*, which was identified to be important for insulin secretion was up-regulated in a mlx mutant background, but down-regulated in a wild-type background (Zinke et al. 2002; Varghese, Lim, and Cohen 2010). This was unexpected as evidence suggests that sugarbabe is directly regulated by the Mondo-Mlx complex and by downstream Dawdle signaling. I have performed RNAi knockdowns for these three transcription factors, and my data suggests that they do play a role in regulating dawdle expression (unpublished). 
Additional factors that are not known to be sugar-responsive include but have been implicated in insulin secretion and metabolism include CG15715 and Adult enhancer factor 1 (aef1). CG15715 is orthologous to human zinc finger protein 706 (ZNF706), which is upregulated in alpha cells in Type I and Type II diabetes (Bosi et al. 2022). ZNF706 is a transcriptional repressor that has been regulate SLCA711 and is largely known to be a potential oncogene in liver cancer (Chu et al. 2024). However, predicted protein-protein interactions (derived from String) suggest that ZNF706 interacts and co-localizes with Thioredoxin domain containing protein-5 (TXNDC5) (Havugimana et al. 2012). Loss of TXNDC5 causes hypersecretion of insulin, why over-expression induces diabetes mellitus (J. Li et al. 2023; Holmgren 1979). Additionally, TXNDC5 catalyzes the reduction of the disulfide bonds of insulin, weakening its ability to bind to its receptor (A. H. Li et al. 2015). Because of its suggested interaction with TXNDC5 and TXNDC5’s effects on insulin, ZNF706’s ortholog CG15715 would be interesting to look at in terms of dawdle regulation. The ring gland is where the equivalent of the alpha cells is found in Drosophila, so identifying where *CG15715* is expressed would be useful to provide additional information in its role in the organism, and if it has impacts on sugar homeostasis.
Aef1 is a transcriptional repressor that is known to bind fat body specific enhancers. This could potentially regulate dawdle expression, as dawdle is heavily expressed and secreted from the fat body of the fly, as well as the brain where it impacts insulin secretion and likewise, the ring gland. The fat body is the human equivalent of the liver and adipose tissue, and is a metabolically important region of the organism. The fat body is a central component to maintain sugar homeostasis, as it is the primary site of glycogen and triglyceride storage, as well as hypoxia and immune responses (Meschi and Delanoue 2021; Huang, Liu, and Perrimon 2022; DiAngelo and Birnbaum 2009). Because *dawdle* is expressed in the fat body, it would be interesting to see if Aef1 regulates dawdle expression under different nutrient conditions. 
Additional factors that were upregulated that are of interest include Modulo (mod). *modulo* encodes a nucleolar protein that is required for growth of proliferative cells through its association with the proto-oncogene Myc (Perrin et al. 2003). Studies have shown that Myc acts downstream of insulin and target of rapamycin (TOR) pathways in Drosophila to regulate growth and proliferation (Parisi et al. 2011). Additionally, in a study using rat cells, c-Myc was required for ChREBP (homolog of Mondo in mammals) to activate glucose-responsive genes (Zhang et al. 2010). It could be that upon insulin secretion and signaling, Modulo could act upon dawdle to either repress or activate its expression in a feedback loop. 
While these factors have not been established to act in a sugar-responsive manner, this data, and its implicated roles in maintaining homeostasis by being implicated in interactions with insulin, activation through downstream insulin pathway components, or being involved in binding to enhancers in a metabolically important organ suggest that they could play a role in maintaining sugar-homeostasis. 
In conclusion, through the identification of differentially expressed genes in a mlx mutant background and wild type background that were challenged on a high sugar diet, I have identified differentially expressed transcription factors that are upregulated. Of these transcription factors, Mondo, Grain, and Sugarbabe are well-established to be sugar-responsive and could potentially regulate dawdle expression. My unpublished data using RNAi knockdowns with these three transcription factors also suggests that they play a role in maintaining sugar homeostasis by regulating dawdle expression. Other factors that may regulate dawdle expression include Aef1, CG15715, and Modulo. Downstream experiments would be to look to see if these genes regulate dawdle by performing RNAi knockdowns, as well as characterizing if these genes are truly sugar-responsive by performing quantitative-PCR on their transcripts on a no-sugar diet and high-sugar diet to validate these findings. 

## Citations
- Berg, Linda van den, Krista Kokki, Sylvia J. Wowro, Konstantin M. Petricek, Onur Deniz, Catrin A. Stegmann, Marius Robciuc, et al. 2023. “Sugar-Responsive Inhibition of Myc-Dependent Ribosome Biogenesis by Clockwork Orange.” *Cell Reports* 42 (7): 112739. [DOI](https://doi.org/10.1016/j.celrep.2023.112739)
- Bosi, Emanuele, Piero Marchetti, Guy Allen Rutter, and Decio Laks Eizirik. 2022. “Human Alpha Cell Transcriptomic Signatures of Types 1 and 2 Diabetes Highlight Disease-Specific Dysfunction Pathways.” *iScience* 25 (10): 105056. [DOI](https://doi.org/10.1016/j.isci.2022.105056)
- Chourpiliadis, Charilaos, and Shamim S. Mohiuddin. 2019. “Biochemistry, Gluconeogenesis.” [NCBI Books](https://www.ncbi.nlm.nih.gov/books/NBK544346/)
- Chu, Jie, Jun Jiang, Xin Fan, Jun Liu, Ke Gao, Yu Jiang, Mengxuan Li, et al. 2024. “A Novel MYC-ZNF706-SLC7A11 Regulatory Circuit Contributes to Cancer Progression and Redox Balance in Human Hepatocellular Carcinoma.” *Cell Death and Differentiation* 31 (10): 1333–48. [DOI](https://doi.org/10.1038/s41418-024-01139-6)
- Das, Rahul, and Leonard L. Dobens. 2015. “Conservation of Gene and Tissue Networks Regulating Insulin Signalling in Flies and Vertebrates.” *Biochemical Society Transactions* 43 (5): 1057–62. [DOI](https://doi.org/10.1042/BST20150094)
- DiAngelo, Justin R., and Morris J. Birnbaum. 2009. “Regulation of Fat Cell Mass by Insulin in *Drosophila Melanogaster*.” *Molecular and Cellular Biology* 29 (24): 6341–52. [DOI](https://doi.org/10.1128/MCB.00811-09)
- Ghosh, Arpan C., and Michael B. O’Connor. 2014. “Systemic Activin Signaling Independently Regulates Sugar Homeostasis, Cellular Metabolism, and pH Balance in *Drosophila Melanogaster*.” *Proceedings of the National Academy of Sciences* 111 (15): 5729–34. [DOI](https://doi.org/10.1073/pnas.1401050111)
- Graham, P., and L. Pick. 2017. “Drosophila as a Model for Diabetes and Diseases of Insulin Resistance.” *Current Topics in Developmental Biology* 121: 397–419. [DOI](https://doi.org/10.1016/bs.ctdb.2016.12.001)
- Havugimana, Pierre C., G. Traver Hart, Tamás Nepusz, Haixuan Yang, Andrei L. Turinsky, Zhihua Li, Peggy I. Wang, et al. 2012. “A Census of Human Soluble Protein Complexes.” *Cell* 150 (5): 1068–81. [DOI](https://doi.org/10.1016/j.cell.2012.08.009)
- Holmgren, A. 1979. “Thioredoxin Catalyzes the Reduction of Insulin Disulfides by Dithiothreitol and Dihydrolipoamide.” *The Journal of Biological Chemistry* 254 (19): 9627–32. [DOI](https://doi.org/10.1016/S0021-9258(19)43791-5)
- Huang, Kerui, Ying Liu, and Norbert Perrimon. 2022. “Roles of Insect Oenocytes in Physiology and Their Relevance to Human Metabolic Diseases.” *Frontiers in Insect Science* 2 (March): 859847. [DOI](https://doi.org/10.3389/finsc.2022.859847)
- Kokki, Krista, Nicole Lamichane, Anni I. Nieminen, Hanna Ruhanen, Jack Morikka, Marius Robciuc, Bohdana M. Rovenko, Essi Havula, Reijo Käkelä, and Ville Hietakangas. 2021. “Metabolic Gene Regulation by Drosophila GATA Transcription Factor Grain.” *PLoS Genetics* 17 (10): e1009855. [DOI](https://doi.org/10.1371/journal.pgen.1009855)
- Kolb, Hubert, Kerstin Kempf, Martin Röhling, and Stephan Martin. 2020. “Insulin: Too Much of a Good Thing Is Bad.” *BMC Medicine* 18 (1): 224. [DOI](https://doi.org/10.1186/s12916-020-01934-4)
- Li, Alexander H., Alanna C. Morrison, Christie Kovar, L. Adrienne Cupples, Jennifer A. Brody, Linda M. Polfus, Bing Yu, et al. 2015. “Analysis of Loss-of-Function Variants and 20 Risk Factor Phenotypes in 8,554 Individuals Identifies Loci Influencing Chronic Disease.” *Nature Genetics* 47 (6): 640–42. [DOI](https://doi.org/10.1038/ng.3305)
- Li, Jing, Yazhuo Chen, Qingyun Liu, Zhufang Tian, and Yan Zhang. 2023. “Mechanistic and Therapeutic Links between Rheumatoid Arthritis and Diabetes Mellitus.” *Clinical and Experimental Medicine* 23 (2): 287–99. [DOI](https://doi.org/10.1007/s10238-023-00930-x)
- Mattila, Jaakko, Essi Havula, Erja Suominen, Mari Teesalu, Ida Surakka, Riikka Hynynen, Helena Kilpinen, et al. 2015. “Mondo-Mlx Mediates Organismal Sugar Sensing through the Gli-Similar Transcription Factor Sugarbabe.” *Cell Reports* 13 (2): 350–64. [DOI](https://doi.org/10.1016/j.celrep.2015.08.054)
- Mattila, Jaakko, and Ville Hietakangas. 2017. “Regulation of Carbohydrate Energy Metabolism in *Drosophila Melanogaster*.” *Genetics* 207 (4): 1231–53. [DOI](https://doi.org/10.1534/genetics.117.300004)
- Meschi, Eleonora, and Renald Delanoue. 2021. “Adipokine and Fat Body in Flies: Connecting Organs.” *Molecular and Cellular Endocrinology* 533 (111339): 111339. [DOI](https://doi.org/10.1016/j.mce.2020.111339)
- Parisi, Federica, Sara Riccardo, Margaret Daniel, Mahesh Saqcena, Nandini Kundu, Annalisa Pession, Daniela Grifoni, Hugo Stocker, Esteban Tabak, and Paola Bellosta. 2011. “Drosophila Insulin and Target of Rapamycin (TOR) Pathways Regulate GSK3 Beta Activity to Control Myc Stability and Determine Myc Expression In Vivo.” *BMC Biology* 9 (1): 65. [DOI](https://doi.org/10.1186/1741-7007-9-65)
- Perrin, Laurent, Corinne Benassayag, Dominique Morello, Jacques Pradel, and Jacques Montagne. 2003. “Modulo Is a Target of Myc Selectively Required for Growth of Proliferative Cells in *Drosophila*.” *Mechanisms of Development* 120 (6): 645–55. [DOI](https://doi.org/10.1016/S0925-4773(03)00110-9)
- Varghese, Jishy, Sing Fee Lim, and Stephen M. Cohen. 2010. “Drosophila MiR-14 Regulates Insulin Production and Metabolism through Its Target, Sugarbabe.” *Genes & Development* 24 (24): 2748–53. [DOI](https://doi.org/10.1101/gad.1964710)
- Zhang, Pili, Mallikarjurna R. Metukuri, Sharell M. Bindom, Edward V. Prochownik, Robert M. O’Doherty, and Donald K. Scott. 2010. “C-Myc Is Required for the CHREBP-Dependent Activation of Glucose-Responsive Genes.” *Molecular Endocrinology (Baltimore, Md.)* 24 (6): 1274–86. [DOI](https://doi.org/10.1210)
